src/webgpu/api/operation/queue/writeBuffer.spec.ts

```diff
@@ -8,3 +8,89 @@ import { makeTestGroup } from '../../../../common/framework/test_group.js';
 import { GPUTest } from '../../../gpu_test.js';
 
 export const g = makeTestGroup(GPUTest);
+
+g.test('writeBuffer,Ranges')
+  .params([
+    {},
+  ])
+  .fn(async t => {
+    const queue = t.device.defaultQueue;
+
+    function runTest(arrayType : any, testBuffer : boolean) {
+      const elementSize = arrayType.BYTES_PER_ELEMENT;
+      const buffer = t.device.createBuffer({ size: 16 * elementSize, usage: GPUBufferUsage.COPY_DST });
+      let arraySm : any = new arrayType(8);
+      let arrayMd : any = new arrayType(16);
+      let arrayLg : any = new arrayType(32);
```

Comment:
Would it be practical to parameterize over the array size, and doing every check on all sizes? It would require some computation of the shouldError values though, so maybe it wouldn't be a net win.

Reply:
Feels like that ends up being more complex and less readable than what I have now without too much concrete benefit.